src/System.Private.Reflection.Core/src/System/Reflection/Runtime/Assemblies/EcmaFormat/EcmaFormatRuntimeAssembly.cs

```diff
@@ -60,19 +59,18 @@ public sealed override IEnumerable<TypeInfo> DefinedTypes
                 if (ReflectionTrace.Enabled)
                     ReflectionTrace.Assembly_DefinedTypes(this);
 #endif
-                throw new NotImplementedException();
-                /*
-                foreach (QScopeDefinition scope in AllScopes)
+                TypeDefinitionHandleCollection allTypes = MetadataReader.TypeDefinitions;
+
+                bool skipFirstType = true; // The first type is always the module type, which isn't returned by this api.
```

Comment:
The name "skipFirstType" doesn't quite match its meaning - "firstTime" or "firstType" would be more accurate.